### ISSUE TITLE
False positive on numpy code

### ISSUE THREAD

**Issue by hauntsaninja** [astral-sh/ty#1690]
### Summary

```py
import numpy as np
import numpy.typing as npt

def foo(x: npt.NDArray, y: int):
    x / y
```
mypy, pyright, pyrefly, zuban all accept this

```
 error[unsupported-operator]: Operator `/` is unsupported between objects of type `ndarray[Any, dtype[_ScalarType_co]]` and `int`
 --> x.py:5:5
  |
4 | def foo(x: npt.NDArray, y: int):
5 |     x / y
  |     ^^^^^
  |
info: rule `unsupported-operator` is enabled by default
```

### Version

ty a29

**Comment by sharkdp** [comment]
Thank you for reporting this.

The false positive is related to the TODOs for generic type aliases [here](https://github.com/astral-sh/ruff/blob/a6cbc138d273b86a70ed36e13623d079bd368236/crates/ty_python_semantic/resources/mdtest/implicit_type_aliases.md?plain=1#L515-L554) (typing `x` as `npt.NDArray[Any]` makes the error go away, but should be equivalent to `npt.NDArray`).

To get the correct return type, we will also have to to support generic protocols in our typevar solver, I think.
